app/src/test/java/nerd/tuxmobil/fahrplan/congress/schedule/LecturesByRoomTest.kt

```diff
@@ -0,0 +1,72 @@
+package nerd.tuxmobil.fahrplan.congress.schedule
+
+import com.google.common.truth.Truth.assertThat
+import nerd.tuxmobil.fahrplan.congress.models.Lecture
+import org.junit.Test
+
+class LecturesByRoomTest {
+
+    @Test
+    fun `no lectures`() {
+        val lecturesByRoom = LecturesByRoom(emptyList())
+
+        assertThat(lecturesByRoom).isEmpty()
+    }
+
+    @Test
+    fun `one lecture in room zero`() {
+        val roomZeroLecture = Lecture("").apply { roomIndex = 0 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomZeroLecture))
+
+        assertThat(lecturesByRoom).hasSize(1)
+        assertThat(lecturesByRoom[0]).containsExactly(roomZeroLecture)
+    }
+
+    @Test
+    fun `one lecture in room one`() {
+        val roomZeroLecture = Lecture("").apply { roomIndex = 1 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomZeroLecture))
+
+        assertThat(lecturesByRoom).hasSize(1)
+        assertThat(lecturesByRoom[1]).containsExactly(roomZeroLecture)
+    }
+
+    @Test
+    fun `one lecture per room - starting from zero, unordered`() {
+        val roomZeroLecture = Lecture("").apply { roomIndex = 0 }
+        val roomOneLecture = Lecture("").apply { roomIndex = 1 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomOneLecture, roomZeroLecture))
+
+        assertThat(lecturesByRoom).hasSize(2)
+        assertThat(lecturesByRoom[0]).containsExactly(roomZeroLecture)
+        assertThat(lecturesByRoom[1]).containsExactly(roomOneLecture)
+    }
+
+    @Test
+    fun `one lecture per room - starting from one, with sparse list`() {
+        val roomOneLecture = Lecture("").apply { roomIndex = 1 }
+        val roomThreeLecture = Lecture("").apply { roomIndex = 3 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomOneLecture, roomThreeLecture))
+
+        assertThat(lecturesByRoom).hasSize(2)
+        assertThat(lecturesByRoom[1]).containsExactly(roomOneLecture)
+        assertThat(lecturesByRoom[3]).containsExactly(roomThreeLecture)
+    }
+
+    @Test
+    @Suppress("UNNECESSARY_NOT_NULL_ASSERTION")
+    fun `multiple lectures per room - starting from one, with sparse list, unordered - preserves input order`() {
+        val roomOneLecture = Lecture("").apply { roomIndex = 1 }
+        val roomThreeLecture1 = Lecture("31").apply { roomIndex = 3 }
+        val roomThreeLecture2 = Lecture("32").apply { roomIndex = 3 }
+        val roomFourLecture = Lecture("").apply { roomIndex = 4 }
+        val lecturesByRoom = LecturesByRoom(listOf(roomFourLecture, roomThreeLecture2, roomOneLecture, roomThreeLecture1))
+
+        assertThat(lecturesByRoom).hasSize(3)
+        assertThat(lecturesByRoom[1]).containsExactly(roomOneLecture)
+        assertThat(lecturesByRoom[3]).containsExactly(roomThreeLecture1, roomThreeLecture2)
+        assertThat(lecturesByRoom[3]!![0]).isEqualTo(roomThreeLecture2)
+        assertThat(lecturesByRoom[3]!![1]).isEqualTo(roomThreeLecture1)
```

Comment:
:mag: Now that `LecturesByRoom` does not return a nullable list these `!!` can be removed. Also the suppress annotation.